arch/posix/CMakeLists.txt

```diff
@@ -10,7 +10,6 @@ else ()
 endif ()
 
 zephyr_compile_options(
-  -MMD
```

Comment:
These 2 lines seem to be just vestigial from the pre cmake era.
@SebastianBoe Should we remove both this and the `-MP` below?

Reply:
Probably. I'd diff the build directory with and without these flags to verify that this is safe.

The flags were first added in commit 76f76441188df11a6b536f0725a2220ccdc503ee

Reply:
Removed -MP